include/fe/fe_abstract.h

```diff
@@ -150,6 +150,27 @@ class FEAbstract : public ReferenceCountedObject<FEAbstract>
                        const std::vector<Point> * const pts = nullptr,
                        const std::vector<Real> * const weights = nullptr) = 0;
 
+  /**
+   * This re-computes the dual shape function coefficients using CUSTOMIZED qrule.
+   * The dual shape coefficients are utilized when calculating dual shape functions.
+   * This does not support InfFE type.
+   */
+  virtual void reinit_dual_shape_coeffs (const Elem * /*elem*/,
+                                         const std::vector<Point> & /*pts*/,
+                                         const std::vector<Real> & /*JxW*/)
+  {
+    libmesh_error_msg("Dual shape coefficient calculation has not been implemented for this FE type.");
+  }
+
+  /**
+   * This re-computes the dual shape function coefficients using DEFAULT qrule.
+   * This does not support InfFE type.
+   */
+  virtual void reinit_default_dual_shape_coeffs (const Elem * /*elem*/)
+  {
+    libmesh_error_msg("Dual shape coefficient calculation has not been implemented for this FE type.");
```

Comment:
Do not all classes go through FE? Just wondering why this is a run time error as opposed to pure virtual

Reply:
Ah I see that InfFE does not. Still I wonder if it would be nicer to have this be pure virtual and put the not implemented error in InfFE? I don’t know what the style preference is for @jwpeterson and @roystgnr

Reply:
I don't have a strong feeling either way for FE since as you mentioned the only other implementation is InfFE

Reply:
I'm happy either way.  We tend to make our most "critical" methods pure virtual, to try to set an example to subclass authors that implementation is something only to be avoided if you can't help it, but the definition of "critical" is a matter of opinion, and especially in this case where we're not expecting any completely new subclasses any time soon the matter seems just academic.

Reply:
Thanks @lindsayad @jwpeterson  @roystgnr for your inputs - Let me make this as pure virtual and add the not implemented error in InfFE

Reply:
Moving it to InfFE does seem to cause some extra test issue. I could not see the details in Civet to make sure whether this is only coverage though. I'll just keep this inside of `fe_base` instead

Reply:
The coverage error has been present for every single completed testing of this PR: https://civet.inl.gov/pr/14815/ so the issue is not this single function. I suspect the orange box is due to the fact that the coverage % of this PR is less than the suggested 90%

Reply:
next time let's not force push so quickly

Reply:
Yea this is from the coverage testing script:
```
          if coverage_passes $COVERAGE_GENERATE_DIFF_NEW_RATE $WARN_NEW; then
            PASSED_COVERAGE+=("New line coverage rate $COVERAGE_GENERATE_DIFF_NEW_RATE%% passes the suggested $WARN_NEW%%")
          else
            WARNED_COVERAGE+=("New new line coverage rate $COVERAGE_GENERATE_DIFF_NEW_RATE%% is less than the suggested $WARN_NEW%%")
            RETURN_CODE=1
```

Reply:
@jwpeterson was right ... even though the timing was very suspiciously correlated with the move to InfFE and back it was not caused by it. @loganharbour pushed a commit fixing the storage logic yesterday afternoon (it had broken last Thursday)

Reply:
Thanks for checking on this! It was such a coincidence that my update was at the same time frame as Logan's.